viewlog.c

```diff
@@ -91,34 +101,54 @@ void ViewLog(int config)
   CloseHandle(proc_info.hProcess);
 }
 
-
-void EditConfig(int config)
+/* either config or config_path */
+void EditConfig(int config, PTCHAR config_path)
 {
-  TCHAR filename[2*MAX_PATH];
+  TCHAR config_file[2*MAX_PATH];
+  TCHAR fullpath[2*MAX_PATH];
+  TCHAR config_dir[2*MAX_PATH];
+  TCHAR assocexe[2*MAX_PATH];
+  DWORD assocexe_num;
 
```

Comment:
size MAX_PATH enough for all paths, 2*MAX_PATH only for the command line. Also eliminate config_file as its only used in the debug print which can use the fullpath instead.